sql/handler.cc

```diff
@@ -550,90 +554,81 @@ int ha_initialize_handlerton(st_plugin_int *plugin)
       hton->discover_table_existence= full_discover_for_existence;
   }
 
-  switch (hton->state) {
-  case SHOW_OPTION_NO:
-    break;
-  case SHOW_OPTION_YES:
-    {
-      uint tmp;
-      ulong fslot;
-
-      DBUG_EXECUTE_IF("unstable_db_type", {
-                        static int i= (int) DB_TYPE_FIRST_DYNAMIC;
-                        hton->db_type= (enum legacy_db_type)++i;
-                      });
-
-      /* now check the db_type for conflict */
-      if (hton->db_type <= DB_TYPE_UNKNOWN ||
-          hton->db_type >= DB_TYPE_DEFAULT ||
-          installed_htons[hton->db_type])
-      {
-        int idx= (int) DB_TYPE_FIRST_DYNAMIC;
+  if (ha_storage_engine_is_enabled(hton))
```

Comment:
I can't immediately see any harm in executing this code for disabled binlog. Could you try removing this condition and running tests?